pkg/operator/k8sutil/pod.go

```diff
@@ -119,6 +120,43 @@ func GetSpecContainerImage(spec v1.PodSpec, name string, initContainer bool) (st
 	return image.Image, nil
 }
 
+// Replaces the pod default toleration of 300s used when the node controller
+// detect a not ready node (node.kubernetes.io/unreachable)
+func AddUnreachableNodeToleration(podSpec *v1.PodSpec) {
+	// The amount of time for this pod toleration can be modified by users
+	// changing the value of <ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE> Rook operator
+	// variable.
+	// Node controller will wait 40 seconds by default before mark a node as
+	// unreachable. After 40s + ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE the pod
+	// will be scheduled in other node
+	// Only one <toleration> to <unreachable> nodes can be added
+	var tolerationSeconds int64 = 5
+	urTolerationDuration := os.Getenv("ROOK_UNREACHABLE_NODE_TOLERATION_LAPSE")
+	if urTolerationDuration != "" {
+		if duration, err := time.ParseDuration(urTolerationDuration); err != nil {
+			logger.Warningf("using default value for <node.kubernetes.io/unreachable> toleration: %v seconds", tolerationSeconds)
+		} else {
+			tolerationSeconds = int64(duration.Seconds())
+		}
+	}
+	urToleration := v1.Toleration{Key: "node.kubernetes.io/unreachable",
+		Operator:          "Exists",
+		Effect:            "NoExecute",
+		TolerationSeconds: &tolerationSeconds}
+
+	var replaced bool
+	for index, item := range podSpec.Tolerations {
+		if item.Key == "node.kubernetes.io/unreachable" {
+			podSpec.Tolerations[index] = urToleration
+			replaced = true
+		}
+		break
+	}
+	if !replaced {
```

Comment:
I think it is possible to eliminate need for the `replaced` var if this function `return`s after finding the toleration and setting it. If it makes it to the end of the loop, it can assume that the toleration was not found and append it.

Reply:
Yes. The suggestion really  improves the code. :-)